crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -230,6 +303,29 @@ impl KeyDecryptable<SymmetricCryptoKey, Vec<u8>> for EncString {
                 EncString::AesCbc256_HmacSha256_B64 { iv, mac, data },
                 SymmetricCryptoKey::Aes256CbcHmacKey(key),
             ) => crate::aes::decrypt_aes256_hmac(iv, mac, data.clone(), &key.mac_key, &key.enc_key),
+            (
+                EncString::XChaCha20_Poly1305_Cose_B64 { data },
+                SymmetricCryptoKey::XChaCha20Poly1305Key(key),
+            ) => {
+                let msg = coset::CoseEncrypt0::from_slice(data.as_slice())
+                    .map_err(|_| CryptoError::EncString(EncStringParseError::InvalidEncoding))?;
+                let decrypted_message = msg
+                    .decrypt(&[], |data, aad| {
+                        let nonce = msg.unprotected.iv.as_slice();
+                        crate::xchacha20::decrypt_xchacha20_poly1305(
+                            nonce.try_into().map_err(|_| CryptoError::EncodingError)?,
+                            key.enc_key
+                                .as_slice()
+                                .try_into()
+                                .expect("XChaChaPoly1305 key is 32 bytes long"),
+                            data,
+                            aad,
+                        )
+                        .map_err(|_| CryptoError::EncodingError)
+                    })
+                    .map_err(|_| CryptoError::EncodingError)?;
+                Ok(decrypted_message)
```

Comment:
Why are we casting all CryptoErrors to EncodingError, and doing it twice?

Is there a valid reason for discarding error information,

Reply:
This is cleaned up, and no longer discarding the error information.